### PR TITLE
Feature/ntgl 261 settings modal

### PR DESCRIPTION
**Pull request by emilytangg**
- Moved existing settings for dark mode and 12h time format to the settings modal
- Added setting to hide classes at full capacity
- Added setting to move newly added classes to the Unscheduled column by default
- Minor changes to styling of settings modal
- Added "compact mode" setting which hides class information

### CODE REVIEW

**Comment by oliverxu42** [comment]
Wait when you select a course there's no more fade in? (Might just be on my end?)

File: client/src/components/Settings.tsx [hunk 1]
```diff
@@ -22,61 +22,104 @@ const CloseButton = styled(IconButton)`
...
-        <StyledDialogTitle disableTypography>
-          <Typography variant="h5">Settings</Typography>
-          <CloseButton aria-label="close" onClick={toggleIsOpen}>
-            <CloseIcon />
-          </CloseButton>
-        </StyledDialogTitle>
-        <Divider />
-        <DialogContent>
-          <Typography variant="body1">
+    const settingsToggles: { state: boolean; setter: (mode: boolean) => void; desc: string }[] = [
```
Comment (munjoonteo) [inline client/src/components/Settings.tsx:76]: We are probably going to do something with all these random type declarations floating around some day... I'd prefer them to be in their own file

File: client/src/components/Settings.tsx [hunk 2]
```diff
@@ -22,61 +22,104 @@ const CloseButton = styled(IconButton)`
...
+
+    const settings = settingsToggles.map((e) => {
+      return (
+        <>
+          <Divider />
+          <ListItem>
             <Switch
-              value={isSquareEdges}
-              checked={isSquareEdges}
+              value={e['state']}
```
Comment (munjoonteo) [inline client/src/components/Settings.tsx]: Can you call this something other than `e` please

File: client/src/components/Settings.tsx [hunk 3]
```diff
@@ -22,61 +22,104 @@ const CloseButton = styled(IconButton)`
...
-        <Divider />
-        <DialogContent>
-          <Typography variant="body1">
+    const settingsToggles: { state: boolean; setter: (mode: boolean) => void; desc: string }[] = [
+      { state: isDarkMode, setter: setIsDarkMode, desc: 'Dark Mode' },
+      { state: isSquareEdges, setter: setIsSquareEdges, desc: 'Square corners on classes' },
+      { state: is12HourMode, setter: setIs12HourMode, desc: 'Display Times in 12-hour format' },
+      { state: isHideFullClasses, setter: setIsHideFullClasses, desc: 'Hide classes that are at full capacity' },
+      { state: isDefaultUnscheduled, setter: setIsDefaultUnscheduled, desc: 'Send newly added classes to Unscheduled by default' },
+      { state: isHideClassInfo, setter: setIsHideClassInfo, desc: 'Hide class information' },
```
Comment (munjoonteo) [inline client/src/components/Settings.tsx]: Hide detailed class information instead?

**Review by munjoonteo — CHANGES_REQUESTED** [review]

**Review by oliverxu42 — APPROVED** [review]
Good stuff!

**Review by munjoonteo — APPROVED** [review]

**Pull request merged**